src/components/render-settings-panel.js

```diff
@@ -486,6 +365,71 @@ class RenderSettingsPanel extends Component {
     buttonHeight: '16px'
   };
 
+  createKeyframe(strCameraType) {    
+    if (this.state.cameraHandle != undefined) { 
+      // NOTE this is where each parts of chain come from
+      // adapter - Deck, scene.animationLoop - Hubble, timeline.detachAnimation - Luma
+      adapter.scene.animationLoop.timeline.detachAnimation(this.state.cameraHandle)
+      adapter.scene.keyframes = this.resetKeyframes() // Resets keyframes so that they don't inherit other options
+      console.log("DETACHED")
+    }
+
+    this.parseSetCameraType(strCameraType, adapter.scene.keyframes.camera)
+
+    const newCameraHandle = adapter.scene.animationLoop.timeline.attachAnimation(adapter.scene.keyframes.camera);
+    this.setState({cameraHandle: newCameraHandle})
+  }
+
+  parseSetCameraType(strCameraType, camera) {
+    const match = strCameraType.match(/\b(?!to)\b\S+\w/g) // returns arr of important keywords. Should work for 2+ words in future ex: ["Orbit", "90"] | ["North", "South"] | ["Zoom", "In"]
+
+    // Named this way for possibility of >2 keyframes in future
+    const firstKeyframe = camera.values[0]
+    const secondKeyframe = camera.values[1]
+
+    // Converts mapState object to turf friendly Point obj (GEOJSON)
+    const point = turf.point([camera.values[1].longitude, camera.values[1].latitude])
+    
+    if (match[0] == "Orbit") {
+      secondKeyframe.bearing = parseInt(match[1])
+    }
+
+    // TODO future option that'll allow user to set X distance (km OR miles) directionally. Options inside turf
+    // https://turfjs.org/docs/#transformTranslate
+    const set_checker = new Set(["East", "South", "West", "North"])
```

Comment:
nit: setChecker

Reply:
Oops that's what you get from coding Python and JS at the same time. Good catch!